source/adios2/helper/adiosCUDAReduceImpl.h

```diff
@@ -197,7 +197,7 @@ __global__ void reduce(const T *__restrict__ g_idata, T *__restrict__ g_odata,
     maskLength = (maskLength > 0) ? (32 - maskLength) : maskLength;
     const unsigned int mask = (0xffffffff) >> maskLength;
```

Comment:
The `mask`, `maskLength`, `ballot_result` variables are not needed anymore so they should be removed.

Reply:
Let me try it on Summit.

Reply:
It was on my side :) 

```bash
[againaru@batch4.summit build]$ ctest -R CUDA
Test project /ccs/home/againaru/adios/ADIOS2-jieyang/build
    Start 883: Engine.BP.*/BPWRZFPCUDA.ADIOS2BPWRZFPCUDA/*.BP4.Serial
1/2 Test #883: Engine.BP.*/BPWRZFPCUDA.ADIOS2BPWRZFPCUDA/*.BP4.Serial ...   Passed    0.92 sec
    Start 884: Engine.BP.*/BPWRZFPCUDA.ADIOS2BPWRZFPCUDA/*.BP4.MPI
2/2 Test #884: Engine.BP.*/BPWRZFPCUDA.ADIOS2BPWRZFPCUDA/*.BP4.MPI ......   Passed    2.47 sec

100% tests passed, 0 tests failed out of 2

Total Test time (real) =   3.64 sec
```

Reply:
I can also confirm that it works on Summit.

Reply:
> The `mask`, `maskLength`, `ballot_result` variables are not needed anymore so they should be removed. Also it's not clear to me why you need a mask of `0xffffffff`

Thanks! I will remove those unused variables. The reason I need a mask of `0xffffffff` is that I need all threads in a warp participating in the thread shuffle operations. Otherwise, If some of the threads are inactive and it happens to be the shuffle targeting thread of an active thread, the returning result is undefined (usually 0), so it makes the final results always 0 (when the true min is positive).